eventuals/event-loop.h

```diff
@@ -844,6 +854,17 @@ inline std::chrono::nanoseconds EventLoop::Clock::Now() {
   if (Paused()) { // TODO(benh): add 'unlikely()'.
     return *paused_ + advanced_;
   } else {
+    CHECK(loop_.in_event_loop_ || !loop_.running_)
+        << "Calling Clock::Now() from a different thread "
+        << "while EventLoop is running will create a data race";
+    // uv_now() doesn't update time inside uv_loop by itself.
+    // Updating it by ourselves will make sure that we return the most
+    // accurate current time.
+    // NOTE(folming): While adding support for asio::io_context
+    // I've noticed that adding io_context.poll() after uv_run() causes
+    // some tests to fail due to Clock::Now() returning an inaccurate value.
```

Comment:
What does "inaccurate" here mean? This comment makes it sound like by changing the time to something "later", all tests pass. But since not all tests were failing in the first place, it sounds like something deterministic is going on that should be figured out, and while adding `uv_update_time()` fixes it, does it actually fix the problem or just mask the problem? If you believe it really fixes the issue, then we really need to articulate what the issue is in this comment.